pkg/providers/vm_provider_interface.go

```diff
@@ -63,4 +63,8 @@ type VirtualMachineProviderInterface interface {
 
 	// VSphereClient returns the provider's vSphere client.
 	VSphereClient(context.Context) (*client.Client, error)
+
+	// DeleteSnapshot deletes a snapshot from a virtual machine.
+	DeleteSnapshot(ctx context.Context, snapshot *vmopv1.VirtualMachineSnapshot,
```

Comment:
Not using virtualmachine.SnapshotArgs as an input here, otherwise will result in import cycle